### PR TITLE
[SPARK-31262][SQL][TESTS] Fix bug tests imported bracketed comments

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR related to https://github.com/apache/spark/pull/27481.
If test case A uses `--IMPORT` to import test case B contains bracketed comments, the output can't display bracketed comments in golden files well.
The content of `nested-comments.sql` show below:
```
-- This test case just used to test imported bracketed comments.

-- the first case of bracketed comment
--QUERY-DELIMITER-START
/* This is the first example of bracketed comment.
SELECT 'ommented out content' AS first;
*/
SELECT 'selected content' AS first;
--QUERY-DELIMITER-END
```
The test case `comments.sql` imports `nested-comments.sql` below:
`--IMPORT nested-comments.sql`
Before this PR, the output will be:
```
-- !query
/* This is the first example of bracketed comment.
SELECT 'ommented out content' AS first
-- !query schema
struct<>
-- !query output
org.apache.spark.sql.catalyst.parser.ParseException

mismatched input '/' expecting {'(', 'ADD', 'ALTER', 'ANALYZE', 'CACHE', 'CLEAR', 'COMMENT', 'COMMIT', 'CREATE', 'DELETE', 'DESC', 'DESCRIBE', 'DFS', 'DROP', 
'EXPLAIN', 'EXPORT', 'FROM', 'GRANT', 'IMPORT', 'INSERT', 'LIST', 'LOAD', 'LOCK', 'MAP', 'MERGE', 'MSCK', 'REDUCE', 'REFRESH', 'REPLACE', 'RESET', 'REVOKE', '
ROLLBACK', 'SELECT', 'SET', 'SHOW', 'START', 'TABLE', 'TRUNCATE', 'UNCACHE', 'UNLOCK', 'UPDATE', 'USE', 'VALUES', 'WITH'}(line 1, pos 0)

== SQL ==
/* This is the first example of bracketed comment.
^^^
SELECT 'ommented out content' AS first


-- !query
*/
SELECT 'selected content' AS first
-- !query schema
struct<>
-- !query output
org.apache.spark.sql.catalyst.parser.ParseException

extraneous input '*/' expecting {'(', 'ADD', 'ALTER', 'ANALYZE', 'CACHE', 'CLEAR', 'COMMENT', 'COMMIT', 'CREATE', 'DELETE', 'DESC', 'DESCRIBE', 'DFS', 'DROP', 'EXPLAIN', 'EXPORT', 'FROM', 'GRANT', 'IMPORT', 'INSERT', 'LIST', 'LOAD', 'LOCK', 'MAP', 'MERGE', 'MSCK', 'REDUCE', 'REFRESH', 'REPLACE', 'RESET', 'REVOKE', 'ROLLBACK', 'SELECT', 'SET', 'SHOW', 'START', 'TABLE', 'TRUNCATE', 'UNCACHE', 'UNLOCK', 'UPDATE', 'USE', 'VALUES', 'WITH'}(line 1, pos 0)

== SQL ==
*/
^^^
SELECT 'selected content' AS first
```
After this PR, the output will be:
```
-- !query
/* This is the first example of bracketed comment.
SELECT 'ommented out content' AS first;
*/
SELECT 'selected content' AS first
-- !query schema
struct<first:string>
-- !query output
selected content
```

### Why are the changes needed?
Golden files can't display the bracketed comments in imported test cases.

### Does this PR introduce any user-facing change?
'No'.

### How was this patch tested?
New UT.

### CODE REVIEW

**Comment by beliefer** [comment]
cc @cloud-fan @maropu

File: sql/core/src/test/resources/sql-tests/inputs/comments.sql [hunk 1]
```diff
@@ -1,12 +1,5 @@
 -- Test comments.
-
--- the first case of bracketed comment
---QUERY-DELIMITER-START
-/* This is the first example of bracketed comment.
-SELECT 'ommented out content' AS first;
-*/
-SELECT 'selected content' AS first;
---QUERY-DELIMITER-END
+--IMPORT nested-comments.sql
```
Comment (maropu) [inline sql/core/src/test/resources/sql-tests/inputs/comments.sql]: What did this means?
```
Golden files can't display well.
```

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala [hunk 2]
```diff
@@ -256,20 +256,23 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
     def splitWithSemicolon(seq: Seq[String]) = {
       seq.mkString("\n").split("(?<=[^\\\\]);")
     }
-    val input = fileToString(new File(testCase.inputFile))
 
-    val (comments, code) = input.split("\n").partition { line =>
+    def splitCommentsAndCodes(input: String) = input.split("\n").partition { line =>
```
Comment (maropu) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala]: Is this fix related to this PR?

File: sql/core/src/test/resources/sql-tests/inputs/comments.sql (hunk 1 above)
Comment (beliefer) [reply sql/core/src/test/resources/sql-tests/inputs/comments.sql]: I updated the description to `Golden files can't display the bracketed comments in imported test cases`.

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala (hunk 2 above)
Comment (beliefer) [reply sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala]: Of course.

File: sql/core/src/test/resources/sql-tests/inputs/comments.sql (hunk 1 above)
Comment (maropu) [reply sql/core/src/test/resources/sql-tests/inputs/comments.sql]: Also, could you put an output example before/after this PR in the PR description?

**Comment by maropu** [comment]
I think this issue is worth filing a new jira.

File: sql/core/src/test/resources/sql-tests/inputs/comments.sql (hunk 1 above)
Comment (beliefer) [reply sql/core/src/test/resources/sql-tests/inputs/comments.sql]: OK. I have put the output example

**Comment by beliefer** [comment]
> I think this issue is worth filing a new jira.

OK. I created a new JIRA and update the title of this PR.

File: sql/core/src/test/resources/sql-tests/inputs/nested-comments.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
+-- This test case just used to test imported bracketed comments.
+
+-- the first case of bracketed comment
+--QUERY-DELIMITER-START
+/* This is the first example of bracketed comment.
+SELECT 'ommented out content' AS first;
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/nested-comments.sql]: This is not a nested comment, right?

Comment (beliefer) [reply sql/core/src/test/resources/sql-tests/inputs/nested-comments.sql]: This is a bracketed comments. 
Maybe I need change `nested-comments.sql` to `imported-comments.sql`?

Comment (cloud-fan) [reply sql/core/src/test/resources/sql-tests/inputs/nested-comments.sql]: maybe updating `SQLQueryTestSuite` only is good enough. It's an obvious bug that we forget to handle `QUERY-DELIMITER` in imported code.

Comment (beliefer) [reply sql/core/src/test/resources/sql-tests/inputs/nested-comments.sql]: You means I shouldn't change `comments.sql` and add `nested-comments.sql` ?

Comment (cloud-fan) [reply sql/core/src/test/resources/sql-tests/inputs/nested-comments.sql]: yea, it's not necessary to separate this test file.

Comment (beliefer) [reply sql/core/src/test/resources/sql-tests/inputs/nested-comments.sql]: OK. I will revert it.

**Review by cloud-fan — APPROVED** [review]

**Review by maropu — APPROVED** [review]

**Comment by maropu** [comment]
Thanks! Merged to master/3.0.

**Comment by beliefer** [comment]
@maropu @cloud-fan @dongjoon-hyun Thanks for all.
